src/services/email/mailgun/mailgun.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import * as FormData from 'form-data';
+import Mailgun, { MailgunClientOptions, MailgunMessageData, MessagesSendResult } from 'mailgun.js';
+
+@Injectable()
+export class MailgunService {
+  private configService: ConfigService = new ConfigService();
+  private mailgun: Mailgun = new Mailgun(FormData);
+  private mailgunClient;
+
+  constructor() {
+    this.mailgunClient = this.mailgun.client({
+      username: 'api',
+      key: this.configService.getOrThrow<string>('MAILGUN_API_KEY'),
+      host: this.configService.getOrThrow<string>('MAILGUN_HOST'),
+    } as MailgunClientOptions);
+  }
+
+  sendEmail(mailgunNotificationData: MailgunMessageData): Promise<MessagesSendResult> {
+    return this.mailgunClient.messages.create(
+      this.configService.getOrThrow<string>('MAILGUN_DOMAIN'),
```

Comment:
No need to call ` this.configService.getOrThrow<string>('MAILGUN_DOMAIN'),` inside function. Keep it as a private string variable like `mailgunDomain` and assign value in constructor. Can be reused if needed later by other service methods.

Reply:
Created new private string variable mailgunDomain and store value from env in it in latest commit.